src/GraphQL/FromServicesAttribute.cs

```diff
@@ -0,0 +1,20 @@
+using GraphQL.Types;
+using GraphQL.Utilities;
+
+namespace GraphQL
+{
+    /// <summary>
+    /// Specifies that the method argument should be pulled from <see cref="IResolveFieldContext.RequestServices"/>.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Parameter)]
+    public class FromServicesAttribute : GraphQLAttribute
+    {
+        /// <inheritdoc/>
+        public override void Modify<TReturnType>(ArgumentInformation<TReturnType> argumentInformation)
```

Comment:
It seems that method is not needed to be generic.

Reply:
No it is not "needed"; I didn't have it generic initially.  But it made the code MUCH more complex and hard to use.  You have to build the expression from scratch and return it.  Not pretty for anyone trying to write their own attribute.

With it as a generic method, writing a custom attribute is trivial, like this one we might even want to include:

```cs
[AttributeUsage(AttributeTargets.Parameter)]
public class FromUserContextAttribute : GraphQLAttribute
{
    public override void Modify<TReturnType>(ArgumentInformation<TReturnType> argumentInformation)
        => argumentInformation.Expression = context => (TReturnType)context.UserContext;
}
```

Reply:
Basically, it hides the complexity for ANY custom attribute that desires to return an expression delegate.

Reply:
> But it made the code MUCH more complex and hard to use. 

Are you 100% sure? Your code or client code? When I looked through your code I had an impression that all that jumps around `TReturnType` are not nessessary. I will not say I'm sure 100%, but intuition tells me that.

Reply:
Also could I use block expression here, i.e. 
```
context =>
{
  Console.WriteLine("afagagsg");
  return (TReturnType)context.UserContext;
}
```
?

Reply:
Client code for custom attributes.

This is the equivalent to the above sample if we do not use generics:

```cs
[AttributeUsage(AttributeTargets.Parameter)]
public class FromUserContextAttribute : GraphQLAttribute
{
    public void Modify(ArgumentInformation argumentInformation)
    {
        var parameter = Expression.Parameter(typeof(IResolveFieldContext));
        var userContextExpression = Expression.MakeMemberAccess(parameter, typeof(IResolveFieldContext).GetProperty(nameof(IResolveFieldContext.UserContext)));
        var castExpression = Expression.Convert(userContextExpression, argumentInformation.ParameterInfo.ParameterType);
        var lambda = Expression.Lambda(castExpression, parameter);
        argumentInformation.Expression = lambda;
    }
}
```

And anything more than a simple property access gets even more complex.  But with generics, you can just write a lambda with C# code and it will compile it into an expression tree for you.

Reply:
Block expressions are not supported by the C# compiler, however, they are supported if you construct them manually.  I have done so before.

![image](https://user-images.githubusercontent.com/6377684/152705907-9bc9a1da-c0e6-48fc-89ab-bb7cc9548e41.png)

Reply:
Here is the method for building a block of code:

https://docs.microsoft.com/en-us/dotnet/api/system.linq.expressions.expression.block?view=net-6.0#system-linq-expressions-expression-block(system-linq-expressions-expression())

You can also write loops and other stuff, if you have the patience for it.

Reply:
Typically it is easier to have the expression call a compiled method and then you write the block of code in there.

![image](https://user-images.githubusercontent.com/6377684/152706046-dc700c35-ad4d-4ad9-9fec-5c83b1d12271.png)

Reply:
Fast coded example:
```
 public interface IResolveFieldContext
    {

    }

    public class ArgumentInformation
    {
        public Expression<Func<IResolveFieldContext, object>> Expression1 { get; set; }

        public LambdaExpression Expression2 { get; set; }
    }




  var i = new ArgumentInformation
        {
            Expression1 = c => "aaa",
            Expression2 = (Expression<Func<IResolveFieldContext, object>>)(c => "aaa"),
        };
```

Just single explicit cast and you get the full power of C# compiler. 
![изображение](https://user-images.githubusercontent.com/21261007/152706257-6d71fc1e-aa56-47dc-9b7c-aa10ef314231.png)

So my idea is that there is no _actual_ need to manipulate `TReturnType`.

Reply:
It can't return a `Func<IResolveFieldContext, object>`.  It must return a `Func<IResolveFieldContext, TParameterType>`.  Without `Modify` as a generic method, it is not possible to write the code you described.

Reply:
And I do not want to cast a returned `object` back to the proper type.  I want it to be typed correctly to begin with.

Reply:
At least with all this discussion I have fixed an incorrect NRT annotation :+1:

Reply:
See #2914